### DISCUSSION TITLE
Does egui support Windows 7?

### DISCUSSION THREAD

**Discussion by 0x7FFFFFFFFFFFFFFF** [emilk/egui#79] [General]
I tried the template project on a Windows 10 box and it works well. Then I copied the generated executable to a Windows 7 box (without Rust insatalled) and it generated the following error. How can I make the executable work on Windows 7?
```
C:\Users\admin\Desktop>egui_template.exe
thread 'main' panicked at 'Failed to compile shader: CompilationError("0:2(10):
error: GLSL 1.40 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.0
0 ES, and 3.00 ES\n", Vertex)', C:\Users\admin\.cargo\git\checkouts\egui-5e4507f
a4153be06\58c025a\egui_glium\src\painter.rs:88:18
stack backtrace:
   0:        0x13fc00059 - <unknown>
   1:        0x13fc154fb - <unknown>
   2:        0x13fbfc8e8 - <unknown>
   3:        0x13fc02df4 - <unknown>
   4:        0x13fc029d8 - <unknown>
   5:        0x13fc036af - <unknown>
   6:        0x13fc03215 - <unknown>
   7:        0x13fc0093f - <unknown>
   8:        0x13fc031c9 - <unknown>
   9:        0x13fc13250 - <unknown>
  10:        0x13fc13093 - <unknown>
  11:        0x13faa795a - <unknown>
  12:        0x13faa1186 - <unknown>
  13:        0x13faa1036 - <unknown>
  14:        0x13faa100c - <unknown>
  15:        0x13fc039f3 - <unknown>
  16:        0x13faa11b7 - <unknown>
  17:        0x13fc1c2a0 - <unknown>
  18:         0x76be652d - BaseThreadInitThunk
  19:         0x76d1c521 - RtlUserThreadStart

C:\Users\admin\Desktop>
```

**Comment by emilk** [comment]
It seems your platform only support GLSL 1.30 and earlier, while egui_glium requires at least GLSL 1.40 (which was released in 2009). Have you tried updating your graphics drivers?

**Reply by 0x7FFFFFFFFFFFFFFF** [reply]
Actually I'm testing in a Windows 7 VM and I tried to update the graphics driver and no updates found. My host computer is running Windows 10 and the graphics driver is the newest.

**Reply by emilk** [reply]
Thanks for testing! I filed an issue here: https://github.com/emilk/egui/issues/80

**Comment by katyo** [comment]
@0x7FFFFFFFFFFFFFFF Can you help me test this pr #187 ?

**Comment by emilk** [comment]
Fixed in https://github.com/emilk/egui/pull/187 (well, I hope!)
